assets/js/base/context/cart-checkout/checkout/processor/index.js

```diff
@@ -1,12 +1,111 @@
+/**
+ * External dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import triggerFetch from '@wordpress/api-fetch';
+import {
+	useCheckoutContext,
+	useShippingDataContext,
+	useBillingDataContext,
+	usePaymentMethodDataContext,
+} from '@woocommerce/base-context';
+import { useEffect, useRef, useCallback } from '@wordpress/element';
+import { useStoreNotices } from '@woocommerce/base-hooks';
+
 /**
  * CheckoutProcessor component. @todo Needs to consume all contexts.
  *
- * Subscribes to checkout context and triggers processing when needed:
- * 		1. Create an order (draft) when email address is provided
- * 		2. Process an order when the place order button is pressed
- * 		   (checkout processing)
+ * Subscribes to checkout context and triggers processing via the API.
  */
 const CheckoutProcessor = () => {
+	const { onCheckoutProcessing, dispatchActions } = useCheckoutContext();
+	const { shippingAddress } = useShippingDataContext();
+	const { billingData } = useBillingDataContext();
+	const { activePaymentMethod } = usePaymentMethodDataContext();
+	const { addErrorNotice } = useStoreNotices();
+	const currentBillingData = useRef( billingData );
+	const currentShippingAddress = useRef( shippingAddress );
+
+	useEffect( () => {
+		currentBillingData.current = billingData;
+		currentShippingAddress.current = shippingAddress;
+	}, [ billingData, shippingAddress ] );
+
+	/**
+	 * Process an order via the /wc/store/checkout endpoint.
+	 *
+	 * @return {boolean} True if everything was successful.
+	 */
+	const processCheckout = useCallback( async () => {
+		await triggerFetch( {
+			path: '/wc/store/checkout',
+			method: 'POST',
+			data: {
+				payment_method: activePaymentMethod,
+				payment_data: [],
+				billing_address: currentBillingData.current,
+				shipping_address: currentShippingAddress.current,
+				customer_note: '',
+			},
+			cache: 'no-store',
+			parse: false,
+		} )
+			.then( ( fetchResponse ) => {
+				// Update nonce.
+				triggerFetch.setNonce( fetchResponse.headers );
+
+				// Handle response.
+				fetchResponse.json().then( function( response ) {
+					if ( ! fetchResponse.ok ) {
+						// We received an error response.
+						if ( response.body && response.body.message ) {
+							addErrorNotice( response.body.message, {
+								id: 'checkout',
+							} );
+						} else {
+							addErrorNotice(
+								__(
+									'Something went wrong. Please check your payment details and try again.',
+									'woo-gutenberg-products-block'
+								),
+								{
+									id: 'checkout',
+								}
+							);
+						}
+					}
+
+					dispatchActions.setRedirectUrl(
+						response.payment_result.redirect_url
+					);
+
+					// @todo do we need to trigger more handling here?
```

Comment:
The checkout processing event emitter responds to what gets returned from subscribers. We can use this for error handling in the followup but basically if you return _anything_ but `true` then the processing aborts on that subscriber and sets the checkout state to `hasError`. The idea is if you return an error object like this:

```jsx
const error  =  {
        message: 'Main error message for top level notice'
        validation: [
             { propertyFailingValidation: 'error message for field' },
             { anotherPropertyFailingValidation: 'error message for field' },
        ]
};
```
Then checkout will  use that both to determine the type of error notices shown and/or any inline validation error notices to show.  This has overlap with the work @Aljullu is doing on #1939 and will likely also require some thought into how payment methods will pick up inline validation errors (or at a minimum we should probably display payment method related errors in context of the payment method step).